game/src/main/java/org/apollo/game/model/entity/obj/GameObjectBounds.java

```diff
@@ -0,0 +1,176 @@
+	private Set<Position> interactionPositions;
+
+	/**
+	 * Creates an EntityBounds.
+	 *
+	 * @param gameObject The GameObject.
+	 */
+	GameObjectBounds(GameObject gameObject) {
+		super(gameObject);
+
+		this.gameObject = gameObject;
+	}
+
+	/**
+	 * Gets the set of positions where the GameObject can be interacted with.
+	 *
+	 * @return The set of positions.
+	 */
+	public Set<Position> getInteractionPositions() {
+		if (interactionPositions == null) {
+			interactionPositions = computeInteractionPositions();
+		}
+
+		return interactionPositions;
+	}
+
+	/**
+	 * Computes the set of positions where the GameObject can be interacted with.
+	 *
+	 * @return The set of positions.
+	 */
+	private Set<Position> computeInteractionPositions() {
+
+		switch (ObjectType.valueOf(gameObject.getType())) {
+			case DIAGONAL_WALL:
+				return getDiagonalWallInteractionPositions();
+
+			case LENGTHWISE_WALL:
+				return getWallInteractionPositions();
+
+			case INTERACTABLE:
+				return getInteractablePositions();
+
+			// TODO: Figure out the rest.
+		}
+
+		return Collections.emptySet();
+	}
+
+	/**
+	 * Gets the set of positions for a diagonal wall where the GameObject can be interacted with.
+	 *
+	 * @return The set of interaction positions.
+	 */
+	private Set<Position> getDiagonalWallInteractionPositions() {
+		Set<Position> positions = new HashSet<>();
+		Direction direction = Direction.WNES[gameObject.getOrientation()];
+		Position position = gameObject.getPosition();
+
+		// TODO: I think this either is missing one, or has one too many. Need to confirm these directions.
+		positions.add(position.step(1, direction));
+		positions.add(position.step(1, direction.counterClockwise()));
+		positions.add(position.step(1, direction.counterClockwise(2)));
+		positions.add(position.step(1, direction.clockwise()));
+		positions.add(position.step(1, direction.clockwise(2)));
+		positions.add(position.step(1, direction.opposite()));
+		positions.add(position.step(1, direction.opposite().counterClockwise()));
+		return positions;
+	}
+
+	/**
+	 * Gets the set of Directions from which the GameObject can be interacted with.
+	 *
+	 * @return The interactable directions.
+	 */
+	private Set<Direction> getInteractableDirections() {
+		Set<Direction> interactableDirections = new HashSet<>();
+
+		int interactionMask = gameObject.getDefinition().getInteractionMask();
+		int rotatedInteractionMask = (interactionMask << gameObject.getOrientation() & 0xf) +
+			(interactionMask >> 4 - gameObject.getOrientation());
+
+		for (Direction direction : Direction.NESW) {
+			boolean interactive = (rotatedInteractionMask & INTERACTION_MASK_BITS.get(direction)) == 0;
+
+			if (interactive) {
+				interactableDirections.add(direction);
+			}
+		}
+
+		return interactableDirections;
+	}
+
+	/**
+	 * Gets the set of positions where the GameObject can be interacted with.
+	 *
+	 * @return The set of interaction positions.
+	 */
+	private Set<Position> getInteractablePositions() {
```

Comment:
I need an example of this in use.  I'm not sure that creating this `Set` all the time is optimal.  Maybe roll this up into your distanced action changes?

Reply:
@garyttierney In the distanced action we provide an `Entity` and check the interaction positions of that entity to decide when to trigger the action for a `Mob`.

Here's an example: https://gist.github.com/stevesoltys/197e7a604f8bfe495ae070c2db0c89c1

Reply:
Note that in my example, I've made the `getInteractionPositions` method abstract in `EntityBounds`.